crates/rspack_core/src/utils/concatenation_scope.rs

```diff
@@ -138,9 +146,13 @@ impl ConcatenationScope {
 
     let mut index_buffer = itoa::Buffer::new();
     let index_str = index_buffer.format(info.index());
-    format!(
+    let module_ref = format!(
       "__WEBPACK_MODULE_REFERENCE__{index_str}_{export_data}{call_flag}{direct_import_flag}{asi_safe_flag}__._"
-    )
+    );
+    let entry = self.refs.entry(*module).or_default();
```

Comment:
We can have a trait for ConcatenationScope for different codegen backend, and put all 'register_import' related into that trait, so the ESMLibraryConcatenationScope can impl the trait and inherit from the original ConcatenationScope